Cleanup SpatialInertia documentation

traversaro opened the pull request:
@prashanthr05 Note that to build the documentation locally you need to run the `dox` target (i.e. `make dox`) and have doxygen installed (`apt install doxygen`).

prashanthr05 commented on src/core/include/iDynTree/Core/SpatialInertiaRaw.h:
```diff
@@ -31,9 +31,9 @@ namespace iDynTree
-        double m_mass; /** mass */
-        double m_mcom[3]; /** first moment of mass (i.e. mass * center of mass */
-        RotationalInertiaRaw m_rotInertia; /** rotational inertia */
+        double m_mass; /** Mass. */
```
These aside comments are passed on as documentation to the next variable declaration. For instance, "Mass" is documented in m_mcom[3].

diegoferigo replied:
Use `///< My comment`

prashanthr05 commented on src/core/include/iDynTree/Core/SpatialInertia.h:
```diff
@@ -84,14 +99,15 @@ namespace iDynTree
-         * Defining \f$ M \f$ as this inertia, return the derivative
-         * with respect to V  of the bias wrench V.cross(M*V).
+         * Defining \f$ M \in \mathbb{R}^{6 \times 6} \f$ as this inertia, return the derivative
+         * with respect to \f$ \mathrm{v} = \begin{bmatrix} v \\ \omega \end{bmatrix} \in \mathbb{R}^6 \f$
```
In the documentation of method `biasWrench(const Twist& v)`, there is a missing identity matrix 1_3 in the 6d inertia matrix defintion.

prashanthr05 replied:
The 6D link velocity is denoted with capital V in that method, which is different from the small v notation used in this method.

traversaro replied:
I am  not getting this comment.

prashanthr05 replied:
Throughout the documentation $m$ is denoted as the mass of the rigid body. So when denoted in the 6D inertia matrix, shouldnt it wrtten as $m1_{3}$, where $1_{3}$ is a $3 \times 3$ identity matrix.

Correct me if I'm wrong.

prashanthr05 commented on src/core/include/iDynTree/Core/SpatialInertia.h:
```diff
@@ -121,12 +137,12 @@ namespace iDynTree
          * |     0    | \f$ m \f$ | The mass of the rigid body |
          * |  2-4     |  \f$ m c \f$      | The first moment of mass of the rigid body |
-         * |  5-9     | \f$ \operatorname{vech}(I_o) \f$ | The 6 indipendent elements of the 3d inertia matrix (\f$ I_{xx} I_{xy} I_{xz} I_{yy} I_{yz} I_{zz} \f$). |
+         * |  5-9     | \f$ \mathop{vech}(I) \f$ | The 6 indipendent elements of the 3d inertia matrix, i.e. \f$ \begin{bmatrix} I_{xx} \\  I_{xy} \\  I_{xz} \\  I_{yy} \\  I_{yz} \\  I_{zz} \end{bmatrix} \f$ . |
```
mc elements 1-3; vech(I) elements 4-9 ?

prashanthr05 replied:
independent*

prashanthr05 commented on src/core/include/iDynTree/Core/SpatialInertia.h:
```diff
@@ -158,14 +174,15 @@ namespace iDynTree
          * \f]
          *
-         * If \f$ \alpha \in \mathbb{R}^10 \f$ is the inertial parameters representation of \f$ I \f$ .
+         * If \f$ \alpha \in \mathbb{R}^10 \f$ is the inertial parameters representation of \f$ M \f$,
```
\mathbb{R}^{10}

prashanthr05 commented on src/core/include/iDynTree/Core/SpatialInertia.h:
```diff
@@ -174,14 +191,15 @@ namespace iDynTree
          * \f]
          *
-         * If \f$ \alpha \in \mathbb{R}^10 \f$ is the inertial parameters representation of \f$ I \f$
+         * If \f$ \alpha \in \mathbb{R}^10 \f$ is the inertial parameters representation of \f$ M \f$,
```
\mathbb{R}^{10}

prashanthr05 commented on src/core/include/iDynTree/Core/SpatialInertia.h:
```diff
@@ -195,17 +213,18 @@ namespace iDynTree
          * Get the matrix
          * \f[
-         *   Y(v,v_r,a_r) \in \mathbb{R}^{6\times6}
+         *   Y(\mathrm{v},\mathrm{v}_r,a_r) \in \mathbb{R}^{6\times6}
```
Y(\mathrm{v},\mathrm{v}_r,a_r) \in \mathbb{R}^{6\times10} ?

prashanthr05 commented on src/core/include/iDynTree/Core/SpatialInertia.h:
```diff
@@ -195,17 +213,18 @@ namespace iDynTree
          * \f]
          *
-         * If \f$ \alpha \in \mathbb{R}^10 \f$ is the inertial parameters representation of \f$ I \f$
+         * If \f$ \alpha \in \mathbb{R}^10 \f$ is the inertial parameters representation of \f$ M \f$, as returned by the
```
\mathbb{R}^{10}

traversaro commented:
@prashanthr05 I should have addressed all comments, except for https://github.com/robotology/idyntree/pull/435#discussion_r180929689 that I am not getting what I should fix.

traversaro commented:
@prashanthr05 Done. Let me know if you can check this in vacation, otherwise I can merge as it is.

prashanthr05 commented:
@traversaro done. You can merge it :)
